src/odata-to-abstract-sql.ts

```diff
@@ -891,7 +923,7 @@ export class OData2AbstractSQL {
 		query.where.push(where);
 	}
 	OrderBy(orderby: OrderByOption, query: Query, resource: Resource) {
-		this.AddExtraFroms(query, resource, orderby.properties);
+		this.AddJoins(query, resource, orderby.properties, 'LeftJoin');
```

Comment:
What's the reasoning for only doing this for `OrderBy`?

Reply:
We probably were only aware of OrderBy adding extra Froms, since queries like `/pilot?$select=was_trained_by__pilot/name` are not something that we are familiar with or use.
Is it only those ^ queries that also add extra froms @Page- (if you remember from cache)?

Reply:
I would be happy to work on this. Should we split that to a separate PR though?

Reply:
Something like this v ?
See: https://github.com/balena-io-modules/odata-to-abstract-sql/pull/167/commits/c44b09d624cb7df9e23b6bb3a70b7b2552cc8c80

I've only replaced FROMs with LEFT JOINs in $select paths, b/c it looked like there was no benefit on doing so for $count & $filter expressions and I was also scared about what that could break, so I would prefer some guidance before touching those.